src/plugins/intel_npu/src/plugin/src/metrics.cpp

```diff
@@ -39,6 +44,13 @@ Metrics::Metrics(const std::shared_ptr<const NPUBackends>& backends) : _backends
                             ov::hint::num_requests.name(),
                             ov::intel_npu::compilation_mode_params.name(),
                             ov::intel_npu::dynamic_shape_to_static.name()};
+
+    // optimizationCapabilities: append dynamic features based on compiler version
+    uint32_t compilerversion = GetCompilerVersion();
+    // Dynamic quantization supported starting from compiler version x
+    if (compilerversion >= ONEAPI_MAKE_VERSION(6, 3)) {
+        _optimizationCapabilities.push_back(std::string("COMPILER_DYNAMIC_QUANTIZATION"));
```

Comment:
I think we should use NPU prefix here as we are sharing the namespace with OV defined capabilities.